cpp/daal/include/services/service_profiler.h

```diff
@@ -0,0 +1,468 @@
+    std::cerr << "Minor version:          " << ver.minorVersion << '\n';
+    std::cerr << "Update version:         " << ver.updateVersion << '\n';
+    std::cerr << "Product status:         " << ver.productStatus << '\n';
+    std::cerr << "Build:                  " << ver.build << '\n';
+    std::cerr << "Build revision:         " << ver.build_rev << '\n';
+    std::cerr << "Name:                   " << ver.name << '\n';
+    std::cerr << "Processor optimization: " << ver.processor << '\n';
+    std::cerr << '\n';
+}
+
+inline void profiler_log_named_args(const char * /*names*/) {}
+
+template <typename T, typename... Rest>
+inline void profiler_log_named_args(const char * names, const T & value, Rest &&... rest)
+{
+    const char * comma = strchr(names, ',');
+    std::string name   = comma ? std::string(names, comma) : std::string(names);
+    name.erase(name.begin(), std::find_if(name.begin(), name.end(), [](unsigned char ch) { return !std::isspace(ch); }));
+    std::cerr << name << ": " << value << "; ";
+    if (comma) profiler_log_named_args(comma + 1, std::forward<Rest>(rest)...);
+}
+
+inline static bool is_service_debug_enabled()
+{
+    static const bool service_debug_value = [] {
+        int value = daal_verbose_mode();
+        return value == 5;
+    }();
+    return service_debug_value;
+}
+
+inline static bool is_logger_enabled()
+{
+    static const bool logger_value = [] {
+        int value = daal_verbose_mode();
+        return value == 1 || value == 4 || value == 5;
+    }();
+    return logger_value;
+}
+
+inline static bool is_tracer_enabled()
+{
+    static const bool verbose_value = [] {
+        std::ios::sync_with_stdio(false);
+        int value = daal_verbose_mode();
+        return value == 2 || value == 4 || value == 5;
+    }();
+    return verbose_value;
+}
+
+inline static bool is_profiler_enabled()
+{
+    static const bool profiler_value = [] {
+        int value = daal_verbose_mode();
+        return value == 1 || value == 2 || value == 3 || value == 4 || value == 5;
+    }();
+    return profiler_value;
+}
+
+inline static bool is_analyzer_enabled()
+{
+    static const bool profiler_value = [] {
+        int value = daal_verbose_mode();
+        return value == 3 || value == 4 || value == 5;
+    }();
+    return profiler_value;
+}
+
+struct task_entry
+{
+    std::int64_t idx;
+    std::string name;
+    std::uint64_t duration;
+    std::int64_t level;
+    std::int64_t count;
+    bool threading_task;
+};
+
+struct task
+{
+    std::vector<task_entry> kernels;
+};
+
+struct task_thread
+{
+    std::set<task_entry> kernels;
+};
+
+class profiler_task
+{
+public:
+    inline profiler_task(const char * task_name, int idx) : task_name_(task_name), idx(idx) {}
+    inline profiler_task(const char * task_name, int idx, bool thread) : task_name_(task_name), idx(idx), is_thread(thread) {}
+    inline ~profiler_task();
+
+private:
+    const char * task_name_;
+    int idx;
```

Comment:
Some private fields have trailing underscore like `task_name_` , other do not like `idx`. Please align the naming.

Reply:
Fixed!